tests/PHPStan/Analyser/nsrt/static-with-this-chained.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace StaticWithThisChained;
+
+use function PHPStan\Testing\assertType;
+
+class Foo
+{
+    /** @var $this */
+    public $propThis;
+
+    /** @var static */
+    public $propStatic;
+
+    /** @var static */
+    public static $propStaticStatic;
+
+    /**
+     * @return $this
+     */
+    public function getThis()
+    {
+        return $this;
+    }
+
+    /**
+     * @return static
+     */
+    public function newStatic()
+    {
+        return new static();
+    }
+
+    /**
+     * @return static
+     */
+    public static function newStaticStatic()
+    {
+        return new static();
+    }
+
+    public function testMethodChained(): void
+    {
+        assertType('$this(StaticWithThisChained\Foo)', $this->getThis());
+        assertType('static(StaticWithThisChained\Foo)', $this->newStatic());
+        assertType('static(StaticWithThisChained\Foo)', static::newStaticStatic());
+        assertType('StaticWithThisChained\Foo', self::newStaticStatic());
+        assertType('StaticWithThisChained\Foo', Foo::newStaticStatic());
+
+        assertType('static(StaticWithThisChained\Foo)', $this->newStatic()->getThis());
```

Comment:
Help needed - how can I mutate the scope to be static when `static` return type is analysed?

Reply:
@ondrejmirtes please see https://phpstan.org/r/29fcfdb5-f19e-4df2-9177-61ad88490c64 repro - do you know where the issue comes from and how it can be fixed? (`static` must never be converted to `$this` )

Reply:
@staabm might I ask you for a help? I have looked into the fix several times, but I am not familiar with mutating the scope logic and I do not know how to fix this myself.